storybook/src/pages/amsterdam/common/AppHeader.tsx

```diff
@@ -1,13 +1,74 @@
-import { Grid, Header, PageMenu } from '@amsterdam/design-system-react'
-import { Alignment as PageMenuStory } from '../../../components/PageMenu/PageMenu.stories'
-
-export const AppHeader = () => (
-  <Grid>
-    <Grid.Cell span="all">
-      <Header
-        links={<PageMenu alignEnd>{PageMenuStory.args?.children}</PageMenu>}
-        menu={<button className="ams-header__menu-button">Menu</button>}
-      />
-    </Grid.Cell>
-  </Grid>
-)
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import { Grid, Header, PageMenu, SearchField } from '@amsterdam/design-system-react'
+import { MenuIcon, SearchIcon } from '@amsterdam/design-system-react-icons'
+import { useMemo, useState } from 'react'
+import { Default as MegaMenuStory } from '../../../components/MegaMenu/MegaMenu.stories'
+
+type Feature = 'menu' | 'search'
+
+export const AppHeader = () => {
+  const [visibleFeature, setVisibleFeature] = useState<Feature | undefined>(undefined)
+
+  const toggleFeature = (feature: Feature) =>
+    setVisibleFeature((currentFeature) => (feature === currentFeature ? undefined : feature))
+
+  const visibilityClass = (feature: Feature) => (feature === visibleFeature ? undefined : 'ams-visually-hidden')
```

Comment:
I just tested this with a screen reader, it's kinda of annoying that it always reads out the mega menu, even if it's collapsed. It also messes up your tab index. I think we should just use `display: none` here, same as gov.uk. Do you have a link to the comment where someone said this was a good idea?

Reply:
The ‘SEO’ comment is kind of [concise](https://codefornl.slack.com/archives/C01DAT4TRPF/p1727947636520699?thread_ts=1727946073.769069&cid=C01DAT4TRPF) :)

Robbert has [more on making header and navigation accessible](https://codefornl.slack.com/archives/C01DAT4TRPF/p1727959649493419?thread_ts=1727946073.769069&cid=C01DAT4TRPF).

Reply:
Ha, that's a little hard to put into context. One thing I can comment on: Using `display: none` is not the same as not having the menu in the DOM. I agree it should be in the DOM, but visually hidden causes too many a11y issues imo.